src/stats.c

```diff
@@ -1477,3 +1492,28 @@ stats_print(void (*write_cb)(void *, const char *), void *cbopaque,
 	emitter_table_printf(&emitter, "--- End jemalloc statistics ---\n");
 	emitter_end(&emitter);
 }
+
+bool
+stats_interval_accum(tsd_t *tsd, uint64_t bytes) {
+	return counter_accum(tsd_tsdn(tsd), &stats_interval_accumulated, bytes);
+}
+
+uint64_t
+stats_interval_accum_batch_size(void) {
+	return stats_interval_accum_batch;
+}
+
+bool
+stats_boot(void) {
+	if (opt_stats_interval > 0) {
+		uint64_t batch = opt_stats_interval >> 6;
+		if (batch > STATS_INTERVAL_ACCUM_BATCH_MAX) {
+			batch = STATS_INTERVAL_ACCUM_BATCH_MAX;
+		} else if (batch == 0) {
+			batch = 1;
+		}
+		stats_interval_accum_batch = batch;
+	}
```

Comment:
Is the purpose of having such a `stats_interval_accum_batch` to trigger the `malloc_stats_print()` more evenly? In other words, if each thread wait for the total interval, then the triggering can be very uneven?

Reply:
Right. The concern would be, if the interval is huge (say, 10G), each thread would wait too long and before it happens.

Reply:
Makes sense. Maybe add some comments? The `>> 6` doesn't seem self-explanatory.

Another thing: if the user specified interval is less than 512 bytes, then it seems we will always trigger the event handler slow path. Maybe we should specify a `STATS_INTERVAL_ACCUM_BATCH_MIN`?

Reply:
Will add comments.

re: batch_min, I don't feel it's necessary, as this is intend to facilitate debugging anyway; if the user sets it to a tiny size, I'd rather we make sure it's triggered accurately to avoid confusion, instead of the overhead (which won't matter much, say stats_print every 128B vs 256B).